test/e2e/protocol/proposer.test.mjs

```diff
@@ -29,10 +38,40 @@ const testProposersUrl = [
   'http://test-proposer4',
 ];
 
+const totalDeposits = 6;
+const nf3User = new Nf3(signingKeys.user1, environment);
+let erc20Address;
+let stateAddress;
+let eventLogs = [];
+
 const web3Client = new Web3Client();
 
 describe('Basic Proposer tests', () => {
   before(async () => {
+    await nf3User.init(mnemonics.user1);
+    erc20Address = await nf3User.getContractAddress('ERC20Mock');
+    stateAddress = await nf3User.stateContractAddress;
+    web3Client.subscribeTo('logs', eventLogs, { address: stateAddress });
+  });
+
+  it('proposer should propose multiple L2 blocks after deposits', async function () {
+    let currentPkdBalance;
```

Comment:
How does this test verify that multiple blocks are generated?

Reply:
Same question too, the rest looks good

Reply:
> During the tests I checked the message sent by `logger.debug('Found ${txDataToSignList.length} blocks to process');` but it's true that I have to pass some check directly to the test and check it there. I've updated the test sending another parameter with the number of blocks proposed in the 'gasCost' event.
> 
> `expect(blocksReceivedToPropose).to.be.equal(Math.floor(totalDeposits / txPerBlock));`

How is `blocksReceivedToPropose` defined ?